bin/subst.ml

```diff
@@ -153,7 +153,13 @@ let subst_file path ~map opam_package_files =
       | None, Some x -> Some x
       | Some x, Some y -> Some (x ^ "\n" ^ y)
     in
-    Option.iter contents ~f:(Io.write_file path)
+    (match contents with
+     | None -> ()
+     | Some contents ->
+       let Unix.{ st_perm; _ } = Path.stat_exn path in
+       Path.chmod path ~mode:(Path.Permissions.add Path.Permissions.write st_perm);
```

Comment:
If the file is already readable, you should skip all this chmoding. Better yet, you should:

1. Try writing
2. If successful, good just move on.
3. If not, then chmod, write, and finally remove the permission.